src/loki/search.cc

```diff
@@ -658,8 +658,15 @@ struct bin_handler_t {
         correlated.edges.erase(new_end, correlated.edges.end());
       }
 
+      //store filtered edges for retry in case find path with correlated.edges failed
+      for (auto filtered_edge : filtered) {
+          // we also add a panelty of 3600 to the score
+          filtered_edge.score = filtered_edge.score + 3600.0f;
```

Comment:
since the filtered_edges are only used on a second pass of the algorithm im not sure we need to increase the score. additonally, why not just move the range into the vector instead of copying them one by one:

    correlated.filtered_edges.insert(correlated.filtered_edges.end(), std::make_move_iterator(filtered.begin()), std::make_move_iterator(filtered.end()));